pip/commands/uninstall.py

```diff
@@ -46,11 +47,11 @@ def run(self, options, args):
         with self._build_session(options) as session:
             format_control = pip.index.FormatControl(set(), set())
             wheel_cache = WheelCache(options.cache_dir, format_control)
+            req_cache = RequirementCache()
```

Comment:
I know this is WIP, but just wanted to point out: This makes me a little uncomfortable, I think either we should have this be a context manager here too so it gets cleaned up or ``RequirementSet`` should take a ``None`` for a ``req_cache``.